### ISSUE TITLE
Update refractor to 4.1.0

### ISSUE THREAD

**Issue by BrendanThompson** [j0lv3r4/mdx-prism#10]
Would it be possible have this using Refractor 4.1.0 as there are a few more languages in there as well as an updated version of Prism.

I did fork this to attempt myself, but unfortunately my JS skills are trash.

**Comment by j0lv3r4** [comment]
@BrendanThompson yeah it's an easy change I think I will do it in the next few days. Unless you want to give it a try and I can help you out.

**Comment by BrendanThompson** [comment]
@j0lv3r4 – I think it would probably best if I just let you do it to be honest.

**Comment by BrendanThompson** [comment]
Hey @j0lv3r4 – I don't suppose you've had any opportunity to do this have you?

**Comment by reesscot** [comment]
Hi @j0lv3r4 I'm interesting in helping get this updated, as `refractor` 4.x includes a security update for `prismjs`. I tried pulling down this repo and bumping `refractor` but when I ran the `yarn test` it fails:

```bash
 FAIL  ./test.js
  ● Test suite failed to run

    /Volumes/workplace/mdx-prism/node_modules/refractor/index.js:8
    export {refractor} from './lib/common.js'
    ^^^^^^

    SyntaxError: Unexpected token 'export'
```

Examing the `refractor` codebase, it looks like they moved from CommonJS modules to ESM when they bumped to 4.x. How would you like to support an ESM import in this repo?

**Comment by juanbzz** [comment]
~~@reesscot sure!~~

Sorry, I misunderstood what you said. I have no idea honestly. do you have any recommendations? I don't have experience moving a library to ESM.

**Comment by juanbzz** [comment]
@reesscot OK, so I have an idea of how to do it. I think we need to do a major release and move entirely to ESM. 

The reason is that our main dependency already did it, so I guess it makes sense for us to do it. Even if we tried to keep it a dual CJS/ESM module, the main dependency has already moved on.

Resources:

* [Get Ready For ESM](https://blog.sindresorhus.com/get-ready-for-esm-aa53530b3f77) by Sindre Sorhus
* [ESM doesn't need to break the ecosystem ](https://dev.to/bcoe/esm-doesn-t-need-to-break-the-ecosystem-4p8b) by Benjamin E. Coe

**Comment by NOI-lgtm** [comment]
Hi @j0lv3r4 I would to point out that there is a high security risk due to this refractor package. 

<img width="791" alt="Screenshot 2022-11-16 at 18 26 12" src="https://user-images.githubusercontent.com/1160546/202250793-9baa1a69-7109-43bd-a38e-7a4b8a9b3386.png">
